### PR TITLE
Issue/7996 actionable empty states (Part 4)

### PR DESCRIPTION
**Pull request by theck13**
### Fix
Add actionable empty states to the search instances as described in https://github.com/wordpress-mobile/WordPress-Android/issues/7996.

#### Media
1. Go to ***My Site*** tab.
2. Tap ***Media*** item under ***Publish*** section.
3. Tap ***Search*** action in toolbar.
4. Enter query not matching any results.
5. Notice actionable empty state shown below.

aes_search_media

#### Media (Stock)
1. Go to ***My Site*** tab.
2. Tap ***Media*** item under ***Publish*** section.
3. Tap ***New Media*** action in toolbar.
4. Tap ***Choose from Free Photo Library*** item in menu.
5. Enter query not matching any results.
6. Notice actionable empty states shown below.

aes_search_media_stock

#### Reader
1. Go to ***Reader*** tab.
2. Tap ***Search*** action in toolbar.
3. Notice actionable empty state shown below.

aes_search_reader

#### Themes
1. Go to ***My Site*** tab.
2. Tap ***Themes*** item under ***Look and Feel*** section.
3. Tap ***Search*** action in toolbar.
4. Enter query not matching any results.
5. Notice actionable empty state shown below.

aes_search_themes

### Review
Only one develop and one designer are required to review these changes, but anyone can perform the review.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/ActionableEmptyView.kt [hunk 1]
```diff
@@ -33,7 +33,7 @@ class ActionableEmptyView : LinearLayout {
     }
```
Comment (AmandaRiu) [inline WordPress/src/main/java/org/wordpress/android/ui/ActionableEmptyView.kt]: Not really related to this PR but I'm just catching up on all the great work done in this series and thought I would make a comment anyway :)
One of the nice things you can do with Kotlin is overload constructors, so you can simplify this class and get rid of several lines of code by doing: 
```
class ActionableEmptyView @JvmOverloads constructor(
    context: Context, 
    attrs: AttributeSet? = null, 
    defStyleAttr: Int = 0
): LinearLayout(context, attrs, defStyleAttr) {
    init {
        initView(context, attrs)
    }
...
}
```

Comment (theck13) [reply WordPress/src/main/java/org/wordpress/android/ui/ActionableEmptyView.kt]: Thanks for the tip, @AmandaRiu!  I used that trick when developing the class originally while I was toying with custom constructors.  I removed those eventually, but never went back to optimize the default ones.  I'll update the `ActionableEmptyView` class in Part 5 of this pull request series.

**Comment by SylvesterWilmott** [comment]
Looks great @theck13

**Comment by theck13** [comment]
Thanks for catching those oddities, @khaykov!

> Stock photos AEV disappears on configuration change

I added a flag to show/hide the empty view on device rotation in 03c6ee2.

> PTR indicator overlaps the AEV label

> AEV is not visible when switching to sites tab in Reader search (not sure this is a part of this PR)

After looking at those two, I'm not sure we need _any_ text while searching since the `SwipeRefreshLayout` animation indicates something is happening.  What do you think, @SylvesterWilmott?

> Not really realted to this PR, but I think we need to figure out when we allow PTR during empty state and when we do not. We disable PRT in Reader search when nothing found, but allow it in Themes search when nothing found

I agree.  List behavior is not consistent across the app.  That includes things like loading/refreshing indicators too.  It is out of the scope of this project, but something we should address.

**Comment by SylvesterWilmott** [comment]
>I'm not sure we need any text while searching since the SwipeRefreshLayout animation indicates something is happening.

Definitely do not need that `Searching...` label, I never noticed this!

>I agree. List behavior is not consistent across the app. That includes things like loading/refreshing indicators too. It is out of the scope of this project, but something we should address.

Also agree here

**Pull request merged**